### PR TITLE
chore(deps): upgrade to go-ipfs 0.5.0

### PR DESCRIPTION
**Pull request by b5**
part of a set of PRs for qri-io/qri#1225

This adds a `Migrate` function for ipfs filestore that does repo migrations. The migration binary itself is separately downloaded, which keeps import overhead small.

### CODE REVIEW

**Comment by b5** [comment]
@ramfox I need to clean up a commented-out test file, but this should be ready to start review. Fairly straightforward.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,13 +1,11 @@
-version: '2'
+version: 2
 jobs:
   build:
     working_directory: /go/src/github.com/qri-io/qfs
     docker:
-      - image: circleci/golang:1.12
+      - image: circleci/golang:latest
```
Comment (Arqu) [inline .circleci/config.yml]: should we really do latest, typically we try to stick to 1.13

**Review by Arqu — APPROVED** [review]
LGTM

**Review by ramfox — APPROVED** [review]
LGTM

**Pull request merged**